gnovm/pkg/gnolang/realm.go

```diff
@@ -453,10 +456,18 @@ func (rlm *Realm) incRefCreatedDescendants(store Store, oo Object) {
 				// NOTE: may already be marked for first gen
 				// newCreated or updated.
 				child.SetOwner(oo)
-				rlm.incRefCreatedDescendants(store, child)
+
+				// There’s a cycle when incrementing the reference count for:
+				// package block -> file block -> package block.
```

Comment:
How does this happen? Shouldn't the Package block already have a non-zero ObjectID and thus be caught by the "recurse guard"?

Reply:
package block as child of fileBlock(unreal), it's refCount > 1 and won't step into next recurse, but the `else if rc > 1 {}` clause. yeah the comment should be improved.

the flow follows: packageValue(unreal)-> packageBlock(unreal) -> fileBlock(unreal) -> packageBlock(Real) -> MarkDirty(packageBlock) -> SetNewReal(packageBlock).

Reply:
updated.